features/cellular/framework/targets/UBLOX/AT/UBLOX_AT.cpp

```diff
@@ -92,32 +92,24 @@ nsapi_error_t UBLOX_AT::init()
 {
     _at->lock();
     _at->flush();
-    _at->cmd_start("AT");
-    _at->cmd_stop_read_resp();
+    _at->at_cmd_discard("", "");
 
 #ifdef TARGET_UBLOX_C027
-    _at->cmd_start("AT+CFUN=0");
-    _at->cmd_stop_read_resp();
+    _at->at_cmd_discard("+CFUN", "=0");
+
     if (_at->get_last_error() == NSAPI_ERROR_OK) {
```

Comment:
"_at->get_last_error()" looks outside lock/unlock as "_at->at_cmd_discard" is unlocking inside it.